express/index.cjs

```diff
@@ -346,9 +346,328 @@ app.post('/login/employer', async (req, res) => {
   }
 });
 
+
+//fetch jobs table from database
+app.get('/api/jobs', async (req, res) => {
+  try {
+    const [jobs] = await req.db.query('SELECT * FROM job');
+    res.status(200).json({ success: true, data: jobs });
+  } catch (err) {
+    console.error('Error fetching jobs:', err);
+    res.status(500).json({ success: false, error: 'Failed to fetch jobs' });
+  }
+});
+
+//forget password, should send email to link
+app.post('/forgot-password', async (req, res) => {
+  const newTime = new Date(Date.now());
+  writer.write(`email sent at : ${setTimestamp(newTime)} \ `);
+  const { email } = req.body;
+
+  try {
+    if (!email) {
+      return res.status(400).json({ success: false, error: 'Missing email' });
+    }
+
+    // Check if the user exists in seekers or employers table
+    const [seeker] = await req.db.query(`SELECT seeker_id FROM Seeker WHERE email = :email`, { email });
+    const [employer] = await req.db.query(`SELECT employer_id FROM Employer WHERE email = :email`, { email });
+
+    if (!seeker.length && !employer.length) {
+      return res.status(404).json({ success: false, error: 'Email not found' });
+    }
+
+    // Generate reset token and send email
+    const resetToken = jwt.sign({ email }, process.env.JWT_SECRET, { expiresIn: '1h' });
+    const resetLink = `http://localhost:5173/reset-password?token=${resetToken}`;
+
+    // Send email (assuming sendEmail is a function defined in Email.js)
+    await sendEmail(email, 'Password Reset', `Click here to reset your password: ${resetLink}`);
+
+    res.status(200).json({ success: true, message: 'Reset link sent to email', resetToken });
+  } catch (err) {
+    console.error(err);
+    res.status(500).json({ success: false, error: 'Server failure' });
+  }
+});
+
+//reset password
+app.put('/reset-password', async (req, res) => {
+  const { token, newPassword } = req.body;
+  const newTime = new Date(Date.now());
+
+  try {
+    if (!token || !newPassword) {
+      return res.status(400).json({ success: false, error: 'Missing token or new password' });
+    }
+
+    const decoded = jwt.verify(token, process.env.JWT_SECRET);
+    const userId = decoded.id;
+    const userType = decoded.type;
+
+    console.log(`Decoded JWT: userId=${userId}, userType=${userType}`);
+
+    const hash = await bcrypt.hash(newPassword, 10);
+
+    if (userType === 'seeker') {
+      query = 'UPDATE Seeker SET user_pass = :hash WHERE seeker_id = :userId';
+      params = { hash, userId };
+    } else {
+      query = 'UPDATE Employer SET user_pass = :hash WHERE employer_id = :userId';
+      params = { hash, userId };
+    }
+
+    console.log(`Executing query: ${query} with params: ${JSON.stringify(params)}`);
+
+    await req.db.query(query, params);
+
+    writer.write(`${setTimestamp(newTime)} | Password reset for ${userId}\n`);
```

Comment:
Log should contain status and req.socket.remoteAddress. Will post log formatting.